File: GamePlayBackend/src/main/kotlin/com/gameplay/gameplaybackend/models/GameModel.kt
package com.gameplay.gameplaybackend.models

import jakarta.persistence.*

@Entity
@Table(name= "games")
class GameModel {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id:Long = 0
    @Column(nullable = false, name = "name")
    var name:String = ""
    @Column(nullable = false, name = "description")
    var description:String = ""
    @Column(nullable = false, name =  "image_url")
    var imageUrl: String = ""
    @Column(nullable = false, name =  "banner_url")
    var bannerUrl: String = ""

}

File: gameplay_app/lib/src/features/home/domain/models/game_model.dart
class GameModel {
  int id;
  String name;
  String imageUrl;
  String bannerUrl;

  GameModel({
    required this.id,
    required this.name,
    required this.imageUrl,
    required this.bannerUrl,
  });

  factory GameModel.fromMap(Map<String, dynamic> map) {
    return GameModel(
      id: map['id'],
      name: map['name'],
      imageUrl: map['imageUrl'],
      bannerUrl: map['bannerUrl'],
    );
  }

  Map<String, dynamic> toMap() {
    return {
      'id': id,
      'name': name,
      'imageUrl': imageUrl,
      'bannerUrl': bannerUrl,
    };
  }
}
